app/viewmodels/pagination/PaginationViewModel.scala

```diff
@@ -0,0 +1,130 @@
+      case (None, 1) => messages("pagination.number-of-movements.singular", "<b>1</b>")
+      case (None, x) => messages("pagination.number-of-movements.plural", s"<b>$x</b>")
+    }
+
+  def paginatedSearchResult(searchParam: Option[String] = None)(implicit messages: Messages): String =
+    searchParam match {
+      case Some(value) =>
+        messages("pagination.results.search",
+          s"<b>${results.from}</b>",
+          s"<b>${results.to}</b>",
+          s"<b>${results.count}</b>",
+          value)
+
+      case None =>
+        messages("pagination.results",
+          s"<b>${results.from}</b>",
+          s"<b>${results.to}</b>",
+          s"<b>${results.count}</b>")
+    }
+
+  val pagination: Pagination = Pagination(Some(items), previous, next)
+}
+
+object PaginationViewModel {
+
+  def apply[T <: PaginationViewModel](totalNumberOfItems: Int,
+                                      currentPage: Int,
+                                      numberOfItemsPerPage: Int,
+                                      href: String,
+                                      additionalParams: Seq[(String, String)])
+                                     (f: (MetaData, Option[PaginationLink], Option[PaginationLink], Seq[PaginationItem]) => T)
+                                     (implicit config: AppConfig): T = {
+
+    val results: MetaData = MetaData(totalNumberOfItems, numberOfItemsPerPage, currentPage)
+    val isNumberOfItemsInsideLimit = totalNumberOfItems <= config.numberOfRecordsToDisableNavigationButtonsInPagination
+
+    val hrefWithParams: Int => String = page => additionalParams.foldLeft(s"$href?page=$page") {
+      case (href, (key, value)) => href + s"&$key=$value"
+    }
+
+    val previous: Option[PaginationLink] = if (isNumberOfItemsInsideLimit) {
+      None
+    } else {
+      previousLinkForItemsGreaterThanAcceptedLimit(currentPage, hrefWithParams(currentPage - 1))
+    }
+
+    val next: Option[PaginationLink] = if (isNumberOfItemsInsideLimit) {
+      None
+    } else {
+      nextLinkForItemsGreaterThanAcceptedLimit(results.totalPages, currentPage, hrefWithParams(currentPage + 1))
+    }
+
+    def populateItems(acc: Seq[PaginationItem], pageNo: Int): Seq[PaginationItem] = {
+      (acc, pageNo) match {
+        case (acc, page) if page == 1 || (page >= currentPage - 1 && page <= currentPage + 1) || page == results.totalPages =>
+          acc :+ PaginationItem(href = hrefWithParams(page), number = Some(page.toString),
+            current = Some(page == currentPage))
+
+        case (acc, _) if (acc.lastOption.flatMap(_.ellipsis).contains(true)) => acc
+
+        case _ => acc :+ PaginationItem(ellipsis = Some(true))
+      }
+    }
+
+    val items = (1 to results.totalPages).foldLeft[Seq[PaginationItem]](Nil) {
+      (acc, page) =>
+        if (isNumberOfItemsInsideLimit) {
+          acc :+ PaginationItem(href = hrefWithParams(page), number = Some(page.toString),
+            current = Some(page == currentPage))
```

Comment:
not sure this needs to be on its on line

Reply:
Done, ideally all the parameters must be on separate lines but as that takes the method's no of line beyond 50, hence it was done like that. Kept on one single line but had to break after PaginationItem(